Mage.Server.Plugins/Mage.Player.Human/src/mage/player/human/HumanPlayer.java

```diff
@@ -1398,6 +1397,40 @@ && lookAtFaceDownCard((Card) object, game, useableAbilities.size())) {
         return false;
     }
 
+    private boolean skipButtonActivated() {
+        return passedAllTurns || passedUntilEndStepBeforeMyTurn || passedTurn || passedUntilEndOfTurn
+                || passedUntilNextMain;
+    }
+
+    private boolean quickStopForLands(Game game) {
+        if (skipButtonActivated()) {
+            Battlefield battlefield = game.getBattlefield();
+            for (Permanent land : battlefield.getAllActivePermanents(filterLand, playerId, game)) {
+                if (land.canUseActivatedAbilities(game)) {
+                    String name = land.getName();
+                    if (name.equals("Maze of Ith") || name.equals("Mystifying Maze") || name.equals("Spires of Orazca")) {
```

Comment:
Do not use names in game engine (it's a bad style). Add property/ability/class/interface instead (example: new method in ActivatedAbility).

Reply:
Although I agree this isn't ideal style, Mystifying Maze uses custom effects, Spires of Orazca uses `RemoveFromCombatTargetEffect`, and the other two cards have a bugged implementation (#7617) that can change in the future. There isn't a good common thread between these lands. I could add a variable, like `shouldQuickStopDuringCombat` to abilities or use regex to parse rule text but those solutions are probably equally ugly. Let me know what you think.